From b732b4822091b91c7363197cec79f4d4097c1a31 Mon Sep 17 00:00:00 2001
From: Arkadiusz Bokowy <arkadiusz.bokowy@gmail.com>
Date: Wed, 13 Jul 2016 21:53:20 +0200
Subject: [PATCH] Check for NULL pointer

Creating XRM database from the NULL pointer is not the best idea there. It
has turned out, that the Xlib function does not have any protection against
coding sloppiness...

Fixes #3
---
 src/main.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/main.c b/src/main.c
index 81ff538..9c60b42 100644
--- a/src/main.c
+++ b/src/main.c
@@ -566,7 +566,8 @@ int main(int argc, char **argv) {
         int rv = 0;

         XrmInitialize();
-        XrmDatabase xrdb = XrmGetStringDatabase(XResourceManagerString(display));
+        const char *data = XResourceManagerString(display);
+        XrmDatabase xrdb = XrmGetStringDatabase(data != NULL ? data : "");

         modules.auth->m.loadxrdb(xrdb);
         modules.background->m.loadxrdb(xrdb);